browse/src/components/ClassificationsDropdown.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div @click="$emit('click', $event)" class="classification-container">
+    <LayeredDropdown
+      :options="options"
+      @input="$emit('input', $event)"
+      v-bind:value="value"
+      :disabled="disabled"
+    />
+  </div>
+</template>
+<script lang="ts">
+import api from "@/api";
+import {
+  defineComponent,
+  onMounted,
+  PropType,
+  ref,
+} from "@vue/composition-api";
+import LayeredDropdown, { Option } from "./LayeredDropdown.vue";
+
+export default defineComponent({
+  props: {
+    value: {
+      type: [Array, String] as PropType<string | string[]>,
+      default: () => [],
+    },
+    disabled: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+    exclude: {
+      type: Array as PropType<string[]>,
+      default: () => [],
+    },
+  },
+  components: {
+    LayeredDropdown,
+  },
+  setup(props) {
+    const options = ref<Option>({ label: "", children: [] });
+
+    const getClassifications = async () => {
+      const cached = localStorage.getItem("classifications");
+      if (cached) {
+        const parsed = JSON.parse(cached);
+        const res = await api.classifications.getClassifications(
```

Comment:
Because you're `await`ing here, you're blocking, and so the cache isn't doing what you intend.  Perhaps it's better to handle the result in `.then`, and return and update the cached option if it changes, but let things go ahead with the cached version until then?

Reply:
Agreed and done, that seems like the smarter caching strategy